builders/pallets-precompiles/precompiles/gmp.md

```diff
@@ -0,0 +1,155 @@
+---
+title: GMP Precompile
+description: Learn about the GMP precompile on Moonbeam and how to use it with the Moonbeam Routed Liquidity program provided by bridges like Wormhole.
+keywords: solidity, ethereum, GMP, wormhole, moonbeam, bridge, connected, contracts, MRL
+---
+
+# Interacting with the GMP Precompile
+
+![GMP Precompile Moonbeam Banner](/images/builders/pallets-precompiles/precompiles/gmp/gmp-banner.png)
+
+## Introduction {: #introduction } 
+
+Moonbeam Routed Liquidity refers to Moonbeam’s use case as the port parachain for liquidity into other Polkadot parachains through a combination of chain-agnostic cross-chain protocols and XCMP. 
+
+The GMP precompile acts as an interface for Moonbeam Routed Liquidity, acting as a middleman between bridging protocols and parachains connected to Moonbeam via XCM. Currently the GMP precompile only supports the relaying of liquidity through Wormhole.
+
+The GMP Precompile is located at the following address:  
+
+=== "Moonbase Alpha"
+     ```
+     {{networks.moonbase.precompiles.gmp}}
+     ```
+
+## The GMP Solidity Interface {: #the-gmp-solidity-interface }
+
+[`Gmp.sol`](https://github.com/PureStake/moonbeam/blob/master/precompiles/gmp/Gmp.sol){target=_blank} is a Solidity interface that allows developers to interact with the precompile’s methods:  
+
+- **wormholeTransferERC20**(*bytes memory* vaa) - receives a wormhole bridge transfer VAA, mints tokens via the wormhole token bridge, and forwards the liquidity to the custom payload’s MultiLocation
+
+In practice, it is unlikely that a developer will have to directly interact with the precompile. Relayers interact with the precompile to complete cross-chain actions, so the origin chain that the cross-chain action originates is where the developer has the most responsibility.
+
+The most common instance that someone will have to interact with the precompile is in the case of a recovery, where a relayer doesn’t complete an MRL transaction. A user would have to find the VAA that comes with their origin chain transaction and then manually invoke the wormholeTransferERC20 function.
+
+## Building the Payload for Wormhole
+
+Currently the GMP precompile only supports sending liquidity through Wormhole.  
+
+To send liquidity, on the origin chain, users must invoke the [`transferTokensWithPayload` method](https://book.wormhole.com/technical/evm/tokenLayer.html#contract-controlled-transfer){target=_blank} on the [WormholeTokenBridge smart contract](https://github.com/wormhole-foundation/wormhole/blob/main/ethereum/contracts/bridge/interfaces/ITokenBridge.sol){target=_blank}. It requires a bytes payload, which must be formatted as a SCALE encoded MultiLocation object wrapped within another versioned type.  
```

Comment:
this guide focuses on moonbeam to other parachains correct? and not backwards? might be good to clearly specify the origin chain you're referring to is moonbeam?